static/js/src/openstack/react/components/CostCalculations.test.jsx

```diff
@@ -0,0 +1,148 @@
+        persistentStorage={{ value: 80, error: "" }}
+        supportLevel="fully-managed"
+      />
+    );
+  });
+
+  afterEach(() => {
+    formWrapper.unmount();
+  });
+
+  it("checks totals are correct with default values", () => {
+    expect(calculationsWrapper.find("#hourly-cost").text()).toEqual("$0.0407");
+    expect(calculationsWrapper.find("#total-savings").text()).toEqual(
+      "$907,656"
+    );
+  });
+
+  it("checks totals are correct with updated values", () => {
+    calculationsWrapper = shallow(
+      <CostCalculations
+        instances={{ value: 2000, error: "" }}
+        vcpus={{ value: 8, error: "" }}
+        ephemeralStorage={{ value: 16, error: "" }}
+        ram={{ value: 32, error: "" }}
+        persistentStorage={{ value: 120, error: "" }}
+        supportLevel="supported"
+      />
+    );
+    expect(calculationsWrapper.find("#hourly-cost").text()).toEqual("$0.1496");
+    expect(calculationsWrapper.find("#total-savings").text()).toEqual(
+      "$7,949,548"
+    );
+  });
+
+  it("checks totals are correct with more updated values", () => {
+    calculationsWrapper = shallow(
+      <CostCalculations
+        instances={{ value: 9000, error: "" }}
+        vcpus={{ value: 32, error: "" }}
+        ephemeralStorage={{ value: 64, error: "" }}
+        ram={{ value: 8, error: "" }}
+        persistentStorage={{ value: 210, error: "" }}
+        supportLevel="fully-managed"
+      />
+    );
+    expect(calculationsWrapper.find("#hourly-cost").text()).toEqual("$0.2756");
+    expect(calculationsWrapper.find("#total-savings").text()).toEqual(
+      "$219,403,189"
+    );
+  });
+
+  it("checks instances prop value updates onChange", () => {
+    formWrapper
+      .find("#instances")
+      .simulate("change", { target: { value: 2000, name: "instances" } });
+
+    expect(formWrapper.find("#instances").props().value).toEqual(2000);
+  });
+
+  it("checks vcupus prop value updates onChange", () => {
+    formWrapper
+      .find("#vcpus")
+      .simulate("change", { target: { value: 6, name: "vcpus" } });
+
+    expect(formWrapper.find("#vcpus").props().value).toEqual(6);
+  });
+
+  it("checks ephemeral storage prop value updates onChange", () => {
+    formWrapper
+      .find("#ephemeral-storage")
+      .simulate("change", { target: { value: 12, name: "ephemeralStorage" } });
+
+    expect(formWrapper.find("#ephemeral-storage").props().value).toEqual(12);
+  });
+
+  it("checks ram prop value updates onChange", () => {
+    formWrapper
+      .find("#ram")
+      .simulate("change", { target: { value: 12, name: "ram" } });
+
+    expect(formWrapper.find("#ram").props().value).toEqual(12);
+  });
+
+  it("checks persistent storage prop value updates onChange", () => {
+    formWrapper
+      .find("#persistent-storage")
+      .simulate("change", { target: { value: 90, name: "persistentStorage" } });
+
+    expect(formWrapper.find("#persistent-storage").props().value).toEqual(90);
```

Comment:
These tests are checking the browser works. Maybe you should check in internal values are correct.